### PR TITLE
Snap created

### PR DESCRIPTION
**Pull request by kz6fittycent**
Initial PR xmrig project

* Snap successfully created and published to snapcraft.io

* Will allow auto-updating of xmrig for users, instead requiring repeated downloads of binaries, or recompiling from source
* When commits made to main repo, builds are automated on snapcraft.io
* Updated README to reflect snap installation instructions

### CODE REVIEW

**Comment by Spudz76** [comment]
Should use advanced build using bundled deps, distro dev packages can be unreliable.

**Comment by kz6fittycent** [comment]
I can definitely give that a shot. Building snaps can be pretty tricky.

**Comment by kz6fittycent** [comment]
Okay, that went well enough. Now performing "advanced build". Build is successfully published and running on test machines.

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      snapcraftctl set-version "$(git describe --tags | sed 's/^v//' | cut -d "-" -f1)"
+      
+    build-packages:
+      - build-essential 
+      - git
+      - cmake
+      - automake
+      - libtool
+      - autoconf
+      - libuv1-dev
```
Comment (Spudz76) [inline snap/snapcraft.yaml:66]: Remove all `-dev` packages, not needed (or wanted) when using bundled.

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+license: GPL-3.0
+
+base: core20
+grade: stable
+confinement: strict
+compression: lzo
+
+architectures:
+  - build-on: amd64
+    run-on: [amd64, armhf, arm64, ppc64el, s390x]
```
Comment (Spudz76) [inline snap/snapcraft.yaml:24]: Technically could have `build-on: [amd64, arm64]`

Seems like run-on should have the same list as build-on, ppc64el and s390x definitely unsupported, etc.

File: snap/snapcraft.yaml (hunk 1 above)
Comment (kz6fittycent) [reply snap/snapcraft.yaml:66]: These are build packages, not actually bundled with the snap. Are you saying that the following are not needed during runtime?
```
      - libuv1-dev 
      - libssl-dev 
      - libhwloc-dev
```

File: snap/snapcraft.yaml (hunk 2 above)
Comment (kz6fittycent) [reply snap/snapcraft.yaml:24]: This is snapcraft formatting. I am checking into it's capability to run on other arches.

Comment (kz6fittycent) [reply snap/snapcraft.yaml:24]: Okay, reading this [documentation](https://snapcraft.io/docs/architectures), I will make sure the two architectures you noted above are the only ones; that's assuming that I'm interpreting the language appropriately.

File: snap/snapcraft.yaml (hunk 1 above)
Comment (Spudz76) [reply snap/snapcraft.yaml:66]: Correct, bundled libs are equivalent to those three packages.

**Comment by kz6fittycent** [comment]
@Dusit2546 sorry for the delay. Are you asking me to PR to the master branch?

**Pull request closed without merging**